modules/phase_field/src/kernels/MultiGrainRigidBodyMotion.C

```diff
@@ -0,0 +1,71 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "MultiGrainRigidBodyMotion.h"
+
+template<>
+InputParameters validParams<MultiGrainRigidBodyMotion>()
+{
+  InputParameters params = validParams<GrainRigidBodyMotionBase>();
+  params.addClassDescription("Adds rigid mody motion to grains");
+  return params;
+}
+
+MultiGrainRigidBodyMotion::MultiGrainRigidBodyMotion(const InputParameters & parameters) :
+    GrainRigidBodyMotionBase(parameters)
+{
+}
+
+Real
+MultiGrainRigidBodyMotion::computeQpResidual()
+{
+  RealGradient vadv_total = 0.0;
+  Real div_vadv_total = 0.0;
+  for (unsigned int i = 0; i < _velocity_advection[_qp].size(); ++i)
+  {
+    vadv_total += _velocity_advection[_qp][i];
+    div_vadv_total += _div_velocity_advection[_qp][i];
+  }
+
+  return vadv_total * _grad_c[_qp] * _test[_i][_qp] + div_vadv_total * _c[_qp] * _test[_i][_qp];
+}
+
+Real
+MultiGrainRigidBodyMotion::computeQpJacobian()
+{
+  if (_c_var == _var.number()) //Requires c jacobian
+    return computeQpCJacobian();
+
+  return 0.0;
+}
+
+Real
+MultiGrainRigidBodyMotion::computeQpOffDiagJacobian(unsigned int jvar)
+{
+  if (_c_var == jvar) //Requires c jacobian
+    return computeQpCJacobian();
+
+  return 0.0;
+}
+
+Real
+MultiGrainRigidBodyMotion::computeQpCJacobian()
```

Comment:
This is something introduced to calculate the jacobian for variable C. This is introduced so that the same kernel can be used by split and non-split version of CH kernels.